### PR TITLE
[Merged by Bors] - feat(RingTheory/TensorProduct): the universal property of the tensor product of algebras

### PR DESCRIPTION
**Pull request by eric-wieser**
The construction used for `CliffordAlgebra.ofBaseChange` generalizes into a universal property for building arbitrary algebra morphisms out of the tensor product.

It turns out this is a known result; it is mentioned at https://en.wikipedia.org/wiki/Tensor_product_of_algebras#Further_properties which cites the linked reference, though the version added by this PR is heterobasic like the rest of this file.

This is a generalization of the existing `Algebra.TensorProduct.productLeftAlgHom` to non-commutative rings.

---

- [x] depends on: #7403
 
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by riccardobrasca — COMMENTED** [review]
Do we have the version in the commutative case? It follows from this one of course, but it is worth to have it without mention the commutator.

**Comment by eric-wieser** [comment]
You're right, we did already have the commutative case (twice!). I've redefined it in terms of the non-commutative case.

File: Mathlib/RingTheory/TensorProduct.lean [hunk 1]
```diff
@@ -844,6 +917,24 @@ end Monoidal
...
+/-- If `A`, `B`, `C` are `R`-algebras, `A` and `C` are also `S`-algebras (forming a tower as
+`·/S/R`), then the product map of `f : A →ₐ[S] C` and `g : B →ₐ[R] C` is an `S`-algebra
+homomorphism.
+
+This is just a special case of `Algebra.TensorProduct.lift` for when `C` is commutative. -/
+abbrev productLeftAlgHom (f : A →ₐ[S] C) (g : B →ₐ[R] C) : A ⊗[R] B →ₐ[S] C :=
+  lift f g (fun _ _ => Commute.all _ _)
+#align algebra.tensor_product.product_left_alg_hom Algebra.TensorProduct.productLeftAlgHom
+
+end
```
Comment (eric-wieser) [inline Mathlib/RingTheory/TensorProduct.lean:934]: This was moved from further down the file, with its implementation replaced.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**